test/unit-test/FreeRTOS_DHCP/FreeRTOS_DHCP_utest.c

```diff
@@ -0,0 +1,3657 @@
+/* Include Unity header */
+#include "unity.h"
+
+/* Include standard libraries */
+#include <stdlib.h>
+#include <string.h>
+#include <stdint.h>
+
+#include "mock_FreeRTOS_IP.h"
+#include "mock_FreeRTOS_Sockets.h"
+#include "mock_FreeRTOS_IP_Private.h"
+#include "mock_FreeRTOS_UDP_IP.h"
+#include "mock_FreeRTOS_ARP.h"
+#include "mock_task.h"
+#include "mock_NetworkBufferManagement.h"
+#include "mock_FreeRTOS_DHCP_mock.h"
+
+#include "FreeRTOS_DHCP.h"
+
+#include "FreeRTOS_DHCP_stubs.c"
+#include "catch_assert.h"
+
+#include "FreeRTOSIPConfig.h"
+
+extern Socket_t xDHCPSocket;
+extern DHCPData_t xDHCPData;
+
+const char * pcHostName = "Unit-Test";
+
+void test_xIsDHCPSocket( void )
+{
+    BaseType_t xReturn;
+    struct xSOCKET xTestSocket;
+
+    xDHCPSocket = &xTestSocket;
+
+    /************************************/
+    /* Test by NOT giving DHCP socket. */
+    xReturn = xIsDHCPSocket( NULL );
+    TEST_ASSERT_EQUAL( pdFALSE, xReturn );
+
+    /************************************/
+    /* Test by giving DHCP socket. */
+    xReturn = xIsDHCPSocket( xDHCPSocket );
+    TEST_ASSERT_EQUAL( pdTRUE, xReturn );
+}
+
+void test_eGetDHCPState( void )
+{
+    DHCPData_t xTestData;
+    eDHCPState_t eReturn;
+    int i;
+
+    for( i = 0; i < sizeof( xTestData.eDHCPState ); i++ )
+    {
+        /* Modify the global state. */
+        xDHCPData.eDHCPState = i;
+        eReturn = eGetDHCPState();
+        TEST_ASSERT_EQUAL( i, eReturn );
+    }
+}
+
+NetworkBufferDescriptor_t * pxGlobalNetworkBuffer[ 10 ];
+uint8_t GlobalBufferCounter = 0;
```

Comment:
statics... a few variables down the line as well

Reply:
Done.